FEAT : BDBD-341 실시간 검색어 api 만들기

onikatashi opened the pull request:
### 개요
 - 실시간 검색어 api 만들기

### 변경사항
 - 따로 검색어를 저장하는 api를 만들었습니다. (일단 테스트용으로 따로 만들었다는 것을 pr 올리면서 깨달았습니다. 바로 검색하는 api와 통합하겠습니다.)
 - 검색어 횟수에 따라 정렬하는 인기 검색어 api를 만들었습니다.

### 관련 지라 및 위키 링크
 - [BDBD-341](https://fake-developers.atlassian.net/browse/BDBD-341)

### 리뷰어에게 하고 싶은 말
 - redis 사용법을 익히는데 시간이 많이 들어버렸습니다. 죄송합니다.
 - 생각하는 방법을 코드로 옮기는데 여러움이 있어서 코드가 엉망일테니 상처받지 않을 강도로 많은 PR 부탁드립니다.

joonamin committed:
[FEAT]: Firebase 연동을 위한 설정 파일 추가

CHORE : 의존성 추가 형식 변경 및 불필요한 개행 제거
1. `build.gradle` 의 implementation을 short-form 으로 변경
2. `application.yaml` 의 불필요한 개행 제거

CHORE : `BDBD-354` 코드 스멜 제거

CHORE : `BDBD-354` 코드 스멜 제거 -2

FEAT : redis 연결

CHORE : 리뷰 반영

CHORE : 코드스멜 수정

CHORE : 코드스멜 수정

Feature : `BDBD-358` FirebaseTokenFilter 추가 및 Security Filter Chain에 적용

CHORE, FIX : `BDBD-358` Code Smell 수정 및 버그 픽스

chore: `BDBD-358` 코드 리팩토링

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

FIX : BDBD-364 게시글 등록 API 버그 수정

FIX : BDBD-364 게시글 등록 API 버그 수정

CHORE : 리뷰 수정

FEAT : BDBD-377 백엔드 issueTransition 적용

FEAT : BDBD-377 백엔드 issueTransition 적용

FEAT : BDBD-377 백엔드 issueTransition 적용

feat: `BDBD-370` firebase 로그인/회원가입 API 작성

FEAT : redis 연결

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

FEAT : BDBD-341 실시간 검색어 api 만들기

FEAT : BDBD-341 실시간 검색어 api 만들기

FEAT : BDBD-341 실시간 검색어 api 만들기

temp

refactor: `BDBD-370` firebase OAuth2 login 코드 리팩토링

joonamin committed:
[FEAT]: Firebase 연동을 위한 설정 파일 추가

CHORE : 의존성 추가 형식 변경 및 불필요한 개행 제거
1. `build.gradle` 의 implementation을 short-form 으로 변경
2. `application.yaml` 의 불필요한 개행 제거

CHORE : `BDBD-354` 코드 스멜 제거

CHORE : `BDBD-354` 코드 스멜 제거 -2

FEAT : redis 연결

CHORE : 리뷰 반영

CHORE : 코드스멜 수정

CHORE : 코드스멜 수정

Feature : `BDBD-358` FirebaseTokenFilter 추가 및 Security Filter Chain에 적용

CHORE, FIX : `BDBD-358` Code Smell 수정 및 버그 픽스

chore: `BDBD-358` 코드 리팩토링

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

FIX : BDBD-364 게시글 등록 API 버그 수정

FIX : BDBD-364 게시글 등록 API 버그 수정

CHORE : 리뷰 수정

FEAT : BDBD-377 백엔드 issueTransition 적용

FEAT : BDBD-377 백엔드 issueTransition 적용

FEAT : BDBD-377 백엔드 issueTransition 적용

feat: `BDBD-370` firebase 로그인/회원가입 API 작성

FEAT : redis 연결

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

FEAT : BDBD-341 실시간 검색어 api 만들기

FEAT : BDBD-341 실시간 검색어 api 만들기

FEAT : BDBD-341 실시간 검색어 api 만들기

temp

refactor: `BDBD-370` firebase OAuth2 login 코드 리팩토링

joonamin committed:
[FEAT]: Firebase 연동을 위한 설정 파일 추가

CHORE : 의존성 추가 형식 변경 및 불필요한 개행 제거
1. `build.gradle` 의 implementation을 short-form 으로 변경
2. `application.yaml` 의 불필요한 개행 제거

CHORE : `BDBD-354` 코드 스멜 제거

CHORE : `BDBD-354` 코드 스멜 제거 -2

FEAT : redis 연결

CHORE : 리뷰 반영

CHORE : 코드스멜 수정

CHORE : 코드스멜 수정

Feature : `BDBD-358` FirebaseTokenFilter 추가 및 Security Filter Chain에 적용

CHORE, FIX : `BDBD-358` Code Smell 수정 및 버그 픽스

chore: `BDBD-358` 코드 리팩토링

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

FIX : BDBD-364 게시글 등록 API 버그 수정

FIX : BDBD-364 게시글 등록 API 버그 수정

CHORE : 리뷰 수정

FEAT : BDBD-377 백엔드 issueTransition 적용

FEAT : BDBD-377 백엔드 issueTransition 적용

FEAT : BDBD-377 백엔드 issueTransition 적용

feat: `BDBD-370` firebase 로그인/회원가입 API 작성

FEAT : redis 연결

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

FEAT : BDBD-341 실시간 검색어 api 만들기

FEAT : BDBD-341 실시간 검색어 api 만들기

FEAT : BDBD-341 실시간 검색어 api 만들기

temp

refactor: `BDBD-370` firebase OAuth2 login 코드 리팩토링

joonamin committed:
feat: BDBD-395 Github Super linter적용

BDBD-395 Github Super linter적용

onikatashi committed:
FEAT : BDBD-341 실시간 검색어 api 만들기

onikatashi committed:
FEAT : BDBD-341 실시간 검색어 api 만들기

onikatashi committed:
FEAT : BDBD-341 실시간 검색어 api 만들기

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/service/ProductService.java:
```diff
@@ -359,6 +369,42 @@ public List<ProductEntity> searchProduct(String searchWord, int searchType, long
         }
     }
 
+    public void saveSearchWord(String searchWord) {
```
검색어라는 도메인은 글쓰기 도메인과 별도로 분리하는게 역할 분리 측면에 좋아 보일것 같은데 어떻게 생각하시나요?

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/service/ProductService.java:
```diff
@@ -359,6 +369,42 @@ public List<ProductEntity> searchProduct(String searchWord, int searchType, long
+            }
+        }
+        List<String> keySet = new ArrayList<>(map.keySet());
+        keySet.sort((o1, o2) -> map.get(o2).compareTo(map.get(o1)));
```
리스트로 받아서 정렬하지 않고 처음부터 map대신에 set에 넣었다면 자연스럽게 정렬완료된 상태이지 않을까요?

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/service/ProductService.java:
```diff
@@ -359,6 +369,42 @@ public List<ProductEntity> searchProduct(String searchWord, int searchType, long
+            if (i == listCount) {
+                break;
+            }
+        }
```
한번의 실행을 위해 여러번 레디스를 바라보는게 좋아보이지 않네요ㅠㅠ
1. 주기적(1시간에 한번 혹은 1분에한번)으로 검색어 순위를 만들어서  캐싱해두면 어떨까요?
2. 나중에 극한의 효율을 위해선 Lua script 라는것도 알아보시면 좋을것 같아요!

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/repository/RedisRepository.java:
```diff
@@ -6,9 +6,9 @@
+    RedisTemplate<String, String> redisTemplate;
 
-    RedisRepository(RedisTemplate<String, Object> redisTemplate) {
+    RedisRepository(RedisTemplate<String, String> redisTemplate) {
```
레디스 템플릿의 자료형이 string, string이면 이미 정의된 StringRedisTemplate 라는 자료형을 쓰시는게 깔끔할것 같아요!

konempty reviewed (changes requested):
고생하셨습니다! 리뷰 확인부탁드릴게요 :)

minseonglove commented on src/main/java/com/fakedevelopers/bidderbidder/service/ProductService.java:
```diff
@@ -189,13 +197,15 @@ private List<ProductListDto> makeProductList(int size, long startNumber) {
 
     // 검색어 있을 때 무한스크롤로 상품 리스트 만들기
     private List<ProductListDto> makeProductList(String searchWord, int searchType, int size, long startNumber) {
+        saveSearchWord(searchWord);
```
공백이 들어오거나 최신 리스트를 받아올 땐 어떻게 동작하지

minseonglove replied:
검색어가 공백으로만 들어올 땐 최신 리스트를 뿌려주게 바까주세요

minseonglove commented on src/main/java/com/fakedevelopers/bidderbidder/service/ProductService.java:
```diff
@@ -359,6 +369,42 @@ public List<ProductEntity> searchProduct(String searchWord, int searchType, long
+        keySet.sort((o1, o2) -> map.get(o2).compareTo(map.get(o1)));
+
+        int i = 0;
+        for (String key : keySet) {
```
keySet을 listCount만큼 자르고 시작했으면 쵸큼 더 깔끔했을지도

minseonglove reviewed (approved):
![de407e11f03250c0c0a1630644e4c03d](https://user-images.githubusercontent.com/90435036/181019304-2d0e6f07-83d5-491c-aa0e-043afcc0c982.jpeg)

onikatashi committed:
FEAT : BDBD-341 @Scheduled 추가로 인기 검색어 자동 초기화
CHORE : 리뷰 반영

onikatashi committed:
FEAT : BDBD-341 @Scheduled 추가로 인기 검색어 자동 초기화
CHORE : 리뷰 반영

joonamin reviewed (approved):
고생많으셨습니다!! 😄

minseonglove reviewed (changes requested):
![1454005572-the-voice-will-i-am-awkward](https://user-images.githubusercontent.com/90435036/181396476-4301f78d-e987-4fe4-93cd-b0e472af33a3.gif)

joonamin committed:
feat: firebase token을 이용한 구글 로그인 구현

joonamin committed:
[FEAT]: Firebase 연동을 위한 설정 파일 추가

CHORE : 의존성 추가 형식 변경 및 불필요한 개행 제거
1. `build.gradle` 의 implementation을 short-form 으로 변경
2. `application.yaml` 의 불필요한 개행 제거

CHORE : `BDBD-354` 코드 스멜 제거

CHORE : `BDBD-354` 코드 스멜 제거 -2

FEAT : redis 연결

CHORE : 리뷰 반영

CHORE : 코드스멜 수정

CHORE : 코드스멜 수정

Feature : `BDBD-358` FirebaseTokenFilter 추가 및 Security Filter Chain에 적용

CHORE, FIX : `BDBD-358` Code Smell 수정 및 버그 픽스

chore: `BDBD-358` 코드 리팩토링

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

FIX : BDBD-364 게시글 등록 API 버그 수정

FIX : BDBD-364 게시글 등록 API 버그 수정

CHORE : 리뷰 수정

FEAT : BDBD-377 백엔드 issueTransition 적용

FEAT : BDBD-377 백엔드 issueTransition 적용

FEAT : BDBD-377 백엔드 issueTransition 적용

feat: `BDBD-370` firebase 로그인/회원가입 API 작성

FEAT : redis 연결

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

FEAT : BDBD-341 실시간 검색어 api 만들기

FEAT : BDBD-341 실시간 검색어 api 만들기

FEAT : BDBD-341 실시간 검색어 api 만들기

temp

refactor: `BDBD-370` firebase OAuth2 login 코드 리팩토링

fix: BDBD-370 버그 수정

fix: 인증 로직 수정 및 잘못된 코드 수정

joonamin committed:
[FEAT]: Firebase 연동을 위한 설정 파일 추가

CHORE : 의존성 추가 형식 변경 및 불필요한 개행 제거
1. `build.gradle` 의 implementation을 short-form 으로 변경
2. `application.yaml` 의 불필요한 개행 제거

CHORE : `BDBD-354` 코드 스멜 제거

CHORE : `BDBD-354` 코드 스멜 제거 -2

FEAT : redis 연결

CHORE : 리뷰 반영

CHORE : 코드스멜 수정

CHORE : 코드스멜 수정

Feature : `BDBD-358` FirebaseTokenFilter 추가 및 Security Filter Chain에 적용

CHORE, FIX : `BDBD-358` Code Smell 수정 및 버그 픽스

chore: `BDBD-358` 코드 리팩토링

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

FIX : BDBD-364 게시글 등록 API 버그 수정

FIX : BDBD-364 게시글 등록 API 버그 수정

CHORE : 리뷰 수정

FEAT : BDBD-377 백엔드 issueTransition 적용

FEAT : BDBD-377 백엔드 issueTransition 적용

FEAT : BDBD-377 백엔드 issueTransition 적용

feat: `BDBD-370` firebase 로그인/회원가입 API 작성

FEAT : redis 연결

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

FEAT : BDBD-341 실시간 검색어 api 만들기

FEAT : BDBD-341 실시간 검색어 api 만들기

FEAT : BDBD-341 실시간 검색어 api 만들기

temp

refactor: `BDBD-370` firebase OAuth2 login 코드 리팩토링

fix: BDBD-370 버그 수정

fix: 인증 로직 수정 및 잘못된 코드 수정

joonamin committed:
[FEAT]: Firebase 연동을 위한 설정 파일 추가

CHORE : 의존성 추가 형식 변경 및 불필요한 개행 제거
1. `build.gradle` 의 implementation을 short-form 으로 변경
2. `application.yaml` 의 불필요한 개행 제거

CHORE : `BDBD-354` 코드 스멜 제거

CHORE : `BDBD-354` 코드 스멜 제거 -2

FEAT : redis 연결

CHORE : 리뷰 반영

CHORE : 코드스멜 수정

CHORE : 코드스멜 수정

Feature : `BDBD-358` FirebaseTokenFilter 추가 및 Security Filter Chain에 적용

CHORE, FIX : `BDBD-358` Code Smell 수정 및 버그 픽스

chore: `BDBD-358` 코드 리팩토링

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

FIX : BDBD-364 게시글 등록 API 버그 수정

FIX : BDBD-364 게시글 등록 API 버그 수정

CHORE : 리뷰 수정

FEAT : BDBD-377 백엔드 issueTransition 적용

FEAT : BDBD-377 백엔드 issueTransition 적용

FEAT : BDBD-377 백엔드 issueTransition 적용

feat: `BDBD-370` firebase 로그인/회원가입 API 작성

FEAT : redis 연결

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

FEAT : BDBD-341 실시간 검색어 api 만들기

FEAT : BDBD-341 실시간 검색어 api 만들기

FEAT : BDBD-341 실시간 검색어 api 만들기

temp

refactor: `BDBD-370` firebase OAuth2 login 코드 리팩토링

fix: BDBD-370 버그 수정

fix: 인증 로직 수정 및 잘못된 코드 수정

joonamin committed:
[FEAT]: Firebase 연동을 위한 설정 파일 추가

CHORE : 의존성 추가 형식 변경 및 불필요한 개행 제거
1. `build.gradle` 의 implementation을 short-form 으로 변경
2. `application.yaml` 의 불필요한 개행 제거

CHORE : `BDBD-354` 코드 스멜 제거

CHORE : `BDBD-354` 코드 스멜 제거 -2

FEAT : redis 연결

CHORE : 리뷰 반영

CHORE : 코드스멜 수정

CHORE : 코드스멜 수정

Feature : `BDBD-358` FirebaseTokenFilter 추가 및 Security Filter Chain에 적용

CHORE, FIX : `BDBD-358` Code Smell 수정 및 버그 픽스

chore: `BDBD-358` 코드 리팩토링

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

FIX : BDBD-364 게시글 등록 API 버그 수정

FIX : BDBD-364 게시글 등록 API 버그 수정

CHORE : 리뷰 수정

FEAT : BDBD-377 백엔드 issueTransition 적용

FEAT : BDBD-377 백엔드 issueTransition 적용

FEAT : BDBD-377 백엔드 issueTransition 적용

feat: `BDBD-370` firebase 로그인/회원가입 API 작성

FEAT : redis 연결

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

FEAT : BDBD-341 실시간 검색어 api 만들기

FEAT : BDBD-341 실시간 검색어 api 만들기

FEAT : BDBD-341 실시간 검색어 api 만들기

temp

refactor: `BDBD-370` firebase OAuth2 login 코드 리팩토링

fix: BDBD-370 버그 수정

fix: 인증 로직 수정 및 잘못된 코드 수정

joonamin committed:
[FEAT]: Firebase 연동을 위한 설정 파일 추가

CHORE : 의존성 추가 형식 변경 및 불필요한 개행 제거
1. `build.gradle` 의 implementation을 short-form 으로 변경
2. `application.yaml` 의 불필요한 개행 제거

CHORE : `BDBD-354` 코드 스멜 제거

CHORE : `BDBD-354` 코드 스멜 제거 -2

FEAT : redis 연결

CHORE : 리뷰 반영

CHORE : 코드스멜 수정

CHORE : 코드스멜 수정

Feature : `BDBD-358` FirebaseTokenFilter 추가 및 Security Filter Chain에 적용

CHORE, FIX : `BDBD-358` Code Smell 수정 및 버그 픽스

chore: `BDBD-358` 코드 리팩토링

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

FIX : BDBD-364 게시글 등록 API 버그 수정

FIX : BDBD-364 게시글 등록 API 버그 수정

CHORE : 리뷰 수정

FEAT : BDBD-377 백엔드 issueTransition 적용

FEAT : BDBD-377 백엔드 issueTransition 적용

FEAT : BDBD-377 백엔드 issueTransition 적용

feat: `BDBD-370` firebase 로그인/회원가입 API 작성

FEAT : redis 연결

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

FEAT : BDBD-341 실시간 검색어 api 만들기

FEAT : BDBD-341 실시간 검색어 api 만들기

FEAT : BDBD-341 실시간 검색어 api 만들기

temp

refactor: `BDBD-370` firebase OAuth2 login 코드 리팩토링

fix: BDBD-370 버그 수정

fix: 인증 로직 수정 및 잘못된 코드 수정

joonamin committed:
feat: firebase token을 이용한 구글 로그인 API 구현

joonamin committed:
[FEAT]: Firebase 연동을 위한 설정 파일 추가

CHORE : 의존성 추가 형식 변경 및 불필요한 개행 제거
1. `build.gradle` 의 implementation을 short-form 으로 변경
2. `application.yaml` 의 불필요한 개행 제거

CHORE : `BDBD-354` 코드 스멜 제거

CHORE : `BDBD-354` 코드 스멜 제거 -2

FEAT : redis 연결

CHORE : 리뷰 반영

CHORE : 코드스멜 수정

CHORE : 코드스멜 수정

Feature : `BDBD-358` FirebaseTokenFilter 추가 및 Security Filter Chain에 적용

CHORE, FIX : `BDBD-358` Code Smell 수정 및 버그 픽스

chore: `BDBD-358` 코드 리팩토링

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

FIX : BDBD-364 게시글 등록 API 버그 수정

FIX : BDBD-364 게시글 등록 API 버그 수정

CHORE : 리뷰 수정

FEAT : BDBD-377 백엔드 issueTransition 적용

FEAT : BDBD-377 백엔드 issueTransition 적용

FEAT : BDBD-377 백엔드 issueTransition 적용

feat: `BDBD-370` firebase 로그인/회원가입 API 작성

FEAT : redis 연결

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

BDBD-272 DB 분기 나누기

FEAT : BDBD-341 실시간 검색어 api 만들기

FEAT : BDBD-341 실시간 검색어 api 만들기

FEAT : BDBD-341 실시간 검색어 api 만들기

temp

refactor: `BDBD-370` firebase OAuth2 login 코드 리팩토링

fix: BDBD-370 버그 수정

fix: 인증 로직 수정 및 잘못된 코드 수정

sandbox 테스트용

joonamin committed:
feat: firbase token을 이용한 구글 로그인 API 구현

joonamin committed:
chore: conflict 해결

joonamin committed:
chore: 코드 스멜 제거

joonamin committed:
chore: 코드 스멜 제거-2

onikatashi committed:
CHORE : 리뷰 반영

onikatashi committed:
CHORE : 코드 스멜 수정

onikatashi committed:
CHORE : 코드 스멜 수정

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/repository/RedisRepository.java:
```diff
@@ -1,14 +1,85 @@
+        ZSetOperations<String, String> zSetOperations = redisTemplate.opsForZSet();
+        Double sum;
+        ArrayList<String> rank = new ArrayList<>();
+        SortedSet<Pair<String, Double>> keySet = new TreeSet<>((o1, o2) -> {
```
o1 o2는 직관적이지 않네요 ㅜㅜ

konempty replied:
SortedSet 대신에 Set은 안될까요?

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/service/InitService.java:
```diff
@@ -0,0 +1,19 @@
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.stereotype.Service;
+
+@Service
```
Component여도 되지않나요?

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/service/InitService.java:
```diff
@@ -0,0 +1,19 @@
+import org.springframework.stereotype.Service;
+
+@Service
+public class InitService implements ApplicationListener<ContextRefreshedEvent> {
```
컴포넌트의 역할과 클래스명이 맞지 않는것 같네요 ㅜㅜ

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/service/InitService.java:
```diff
@@ -0,0 +1,19 @@
+@Service
+public class InitService implements ApplicationListener<ContextRefreshedEvent> {
+
+    ProductService productService;
```
private final이여도 괜찮지 않나요?

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/repository/RedisRepository.java:
```diff
@@ -1,14 +1,85 @@
+        Double sum;
+        ArrayList<String> rank = new ArrayList<>();
+        SortedSet<Pair<String, Double>> keySet = new TreeSet<>((o1, o2) -> {
+            if (o2.getSecond() - o1.getSecond() == 0) {
```
참고로 double 연산에서 == 0은 위험한 연산입니다..!

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/repository/RedisRepository.java:
```diff
@@ -1,14 +1,85 @@
+                    return 1;
+                else {
+                    return o2.getFirst().compareTo(o1.getFirst());
+                }
```
여기 if else문이 과연 필요한걸까요 좀더 가벼운 연산으로도 가능할것 같아요!

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/repository/RedisRepository.java:
```diff
@@ -1,14 +1,85 @@
+            for (String keyWord : range) {
+                sum = 0d;
+                for (String valueWord : zSetOperations.range(keyWord, 0, -1)) {
+                    sum += zSetOperations.score(keyWord, valueWord);
```
이 for문은 List<Double> list = zSetOperations.score(keyWord, zSetOperations.range(keyWord, 0, -1).toArray());
이렇게 압축이 가능할것 같아요!

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/config/RedisAutoConfig.java:
```diff
@@ -14,8 +14,8 @@
+        StringRedisTemplate redisTemplate = new StringRedisTemplate();
         redisTemplate.setConnectionFactory(lettuceConnectionFactory);
         redisTemplate.setKeySerializer(new StringRedisSerializer());
         redisTemplate.setValueSerializer(new StringRedisSerializer());
```
StringRedisTemplate를 쓰면 Serializer를 설정해주지 않아도 됐던것 같은데 확인 부탁드릴게요!

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/repository/RedisRepository.java:
```diff
@@ -1,14 +1,85 @@
+    public List<String> getPopularSearchWord() {
+        ZSetOperations<String, String> zSetOperations = redisTemplate.opsForZSet();
+        Double sum;
+        ArrayList<String> rank = new ArrayList<>();
```
List<String> rank =  new ArrayList<>(); 로 축약이 될것 같네요

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/repository/RedisRepository.java:
```diff
@@ -1,14 +1,85 @@
+            return yesterdaySearchRank;
+    }
+
+    public void deleteSearchWords() {
```
함수의 역할은 rank를 만든다, 레디스에서 지운다 이렇게 2개 인데 함수 명은 후자만을 반영하네요..!

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/service/ProductService.java:
```diff
@@ -359,6 +361,17 @@ public List<ProductEntity> searchProduct(String searchWord, int searchType, long
+    @Scheduled(cron = "0 0 0 * * *")
+    private void deleteSearchWords() {
+        redisRepository.deleteSearchWords();
+    }
```
스케쥴러가 서비스에 있는건 역할분리 측면에서 맞지 않는것 같네요..!

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/repository/RedisRepository.java:
```diff
@@ -1,14 +1,85 @@
 public class RedisRepository {
 
-    RedisTemplate<String, Object> redisTemplate;
+    StringRedisTemplate redisTemplate;
```
private final 가능하지 않을까요?

konempty reviewed (approved):
고생하셨습니다! 조금만더 수정 부탁드릴게요..!

onikatashi committed:
CHORE : 리뷰 반영

onikatashi committed:
CHORE : 리뷰 반영

onikatashi committed:
CHORE : 리뷰 반영

onikatashi committed:
CHORE : 리뷰 반영

onikatashi committed:
CHORE : 리뷰 반영

onikatashi committed:
CHORE : 리뷰 반영

onikatashi committed:
CHORE : 리뷰 반영

joonamin committed:
CHORE: 리뷰 반영

joonamin committed:
CHORE: 리뷰 반영_2

onikatashi committed:
CHORE : 리뷰 반영

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/repository/RedisRepository.java:
```diff
@@ -1,14 +1,102 @@
+  }
+
+  public void getSearchWords() {
+    yesterdaySearchRank = getPopularSearchWord();
```
한줄짜리 메소드를 만드신 이유가 있으실까요?
별 다른 이유가 없으시다면 getPopularSearchWord 메소드 안에서 yesterdaySearchRank의 값을 변경해줘도 괜찮을것 같아요!

konempty reviewed (approved):
정말 오랫동안 고생 많으셨습니다! 👍 리뷰 내용만 확인해주시고 문제 없다고 판단하시면 머지하셔도 좋을것 같아요!ㅋㅋㅋㅋ

konempty commented on src/main/java/com/fakedevelopers/bidderbidder/service/InitService.java:
```diff
@@ -0,0 +1,23 @@
+
+  public void onApplicationEvent(ContextRefreshedEvent event) {
+    redisRepository.getSearchWords();
+    productService.getPopularSearchWord(10);
```
productService.getPopularSearchWord()가 맞지 않을까요?

onikatashi committed:
CHORE : 리뷰 반영

onikatashi committed:
CHORE : 리뷰 반영

joonamin committed:
CHORE: 리뷰 반영_3

minseonglove reviewed (approved):
![ZVuTIUk](https://user-images.githubusercontent.com/90435036/183101375-6cc49b68-38e4-4f1a-ae03-71bb94fe5616.gif)

onikatashi committed:
CHORE : 머지 컨플릭트 수정

onikatashi committed:
Delete GetPopularSearchWordAtStart.java

onikatashi committed:
Merge branch 'develop' into feature/BDBD-341
